ui/components/link.js

```diff
@@ -3,24 +3,35 @@ import React from 'react';
 
 import { Link as RouterLink, routes } from '../routes';
 
-export default function Link({ children, href, params, route, ...otherProps }) {
-  if (route) {
-    return (
-      <RouterLink params={params} route={route}>
-        <a {...otherProps}>{children}</a>
-      </RouterLink>
-    );
-  } else if (href) {
-    const props = {
-      ...otherProps,
-      href,
-    };
-    if (href.search(/^https|http|^\/\//) === 0) {
-      props.target = '_blank';
-      props.rel = 'noopener noreferrer';
-      props['aria-label'] = 'Link opens in a new window';
+export default class Link extends React.Component {
+  focus() {
+    if (this.el) {
+      this.el.focus();
     }
-    return <a {...props}>{children}</a>;
+  }
+
+  render() {
+    const { children, href, params, route, ...otherProps } = this.props;
+
+    if (route) {
+      return (
+        <RouterLink params={params} route={route}>
+          <a {...otherProps}>{children}</a>
+        </RouterLink>
+      );
+    } else if (href) {
+      const props = {
+        ...otherProps,
+        href,
+      };
+      if (href.search(/^https|http|^\/\//) === 0) {
+        props.target = '_blank';
+        props.rel = 'noopener noreferrer';
+        props['aria-label'] = 'Link opens in a new window';
+      }
+      return <a {...props} ref={(el) => { this.el = el; }}>{children}</a>;
+    }
+    throw new Error('Link created without an href or route');
```

Comment:
Interesting... The function-based version of this component passed eslint fine, but when converted to a `render()` method eslint complained about the following:

```
/usr/src/app/ui/components/link.js
  13:3  error  Expected to return a value at the end of method 'render'  consistent-return
```

I chose to make it throw an error because this seems like that kind of situation, but alternatively I could change it to log a warning and return `null`, or return a React element that displays a warning on the webpage, or something else. Or, if that branch really _is_ a valid use case, we could just return `null` and not log anything.

Any preferences?

Reply:
Nice. I'd bet the linter not catching this before was a bug. I like the exception approach. We could instead remove the final guard (`else if (href)`), but that'd generate some silent fail cases. I think the same is true for `null`.